test/controllers/api-user.test.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const ApiUserController = require('../../api/controllers/api-user.js')
+const fetch = require('node-fetch')
+
+jest.mock('node-fetch')
+
+describe('ApiUserController', () => {
+  it('should be defined', () => {
+    expect(ApiUserController).toBeDefined()
+  })
+
+  describe('lookupUser', () => {
+    it('should lookup the user via Okta', done => {
+      const guid = 'guid'
+
+      const mockedResponse = {
+        ok: true,
+        status: 200,
+        json: () => Promise.resolve([{ profile: { theKeyGuid: guid } }])
+      }
+      fetch.mockImplementation(() => Promise.resolve(mockedResponse))
+
+      ApiUserController.lookupUser('bill.bright@cru.org').then((foundGuid) => {
+        expect(fetch).toHaveBeenCalledWith('https://signon.okta.com/api/v1/users?search=profile.email%20eq%20%22bill.bright%40cru.org%22', expect.anything())
+        expect(foundGuid).toBe(guid)
+        done()
+      }).catch(done)
+    })
+
+    it('should strip quotation marks from the email address', done => {
+      const guid = 'guid'
+
+      const mockedResponse = {
+        ok: true,
+        status: 200,
+        json: () => Promise.resolve([{ profile: { theKeyGuid: guid } }])
+      }
+      fetch.mockImplementation(() => Promise.resolve(mockedResponse))
+
+      ApiUserController.lookupUser('"bill.bright"@"cru".org').then(() => {
+        expect(fetch).toHaveBeenCalledWith('https://signon.okta.com/api/v1/users?search=profile.email%20eq%20%22bill.bright%40cru.org%22', expect.anything())
+        done()
+      }).catch(done)
+    })
+
+    it('should reject if Okta returns an error', done => {
+      const mockedResponse = {
+        ok: false,
+        status: 500,
+        json: () => Promise.resolve(null)
+      }
+      fetch.mockImplementation(() => Promise.resolve(mockedResponse))
+
+      ApiUserController.lookupUser('bill.bright@cru.org').then(() => {
+        done(new Error('Unexpected success'))
```

Comment:
I think Jest has a `fail()` function if that would work here.

Reply:
Can you link to docs for `fail`? I'm having trouble finding it. I found out that I can at least just do `done('Unexpected success')` without passing it an `Error` object. So I think that `done` might double as a `fail` function if passed an argument.

Reply:
https://jasmine.github.io/api/4.5/global.html#fail

`fail()` is a function of Jasmine, which Jest uses under the hood.

Reply:
At least for the version of Jest we're using (v27.5.1), Jest uses a fork of Jasmine called jest-jasmine2, and I think they must have removed the fail global.

```js
console.log(Object.keys(globalThis))

[
      'global',         'clearInterval',
      'clearTimeout',   'setInterval',
      'setTimeout',     'Buffer',
      'setImmediate',   'clearImmediate',
      'ArrayBuffer',    'Uint8Array',
      'URL',            'URLSearchParams',
      'TextEncoder',    'TextDecoder',
      'queueMicrotask', 'AbortController',
      'AbortSignal',    'Event',
      'EventTarget',    'performance',
      'atob',           'btoa',
      'process',        'console',
      'afterAll',       'afterEach',
      'beforeAll',      'beforeEach',
      'describe',       'it',
      'test',           'fdescribe',
      'fit',            'xdescribe',
      'xit',            'xtest',
      'expect',         '__core-js_shared__',
      '__fixturesDir'
]
```

Reply:
Lame